### PR TITLE
GSAGH-550: Analysis case is not consistent with the Load case in Gsa_GH

### PR DESCRIPTION
**Pull request by SandeepArup**
Now result are availble:

image

### CODE REVIEW

**Review by psarras — CHANGES_REQUESTED** [review]

File: GsaGH/Helpers/Assembly/ModelAssembly.cs [hunk 1]
```diff
@@ -249,26 +248,7 @@ private void CheckIfModelIsEmpty() {
...
-            continue;
-          }
-
-          foreach (GsaAnalysisCase ca in task.Cases) {
-            _model.AddAnalysisCaseToTask(task.Id, ca.Name, ca.Definition);
-          }
-        }
-      }
+      ModelFactory.BuildAnalysisTask(_model, analysisTasks);
     }
```
[the remark this answers is not in the document]
Comment (SandeepArup) [reply GsaGH/Helpers/Assembly/ModelAssembly.cs]: @psarras done.

File: GsaGHTests/Helpers/GsaAPI/ModelFactoryTests.cs [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+using GsaAPI;
+
+using GsaGH.Helpers;
+using GsaGH.Helpers.GsaApi;
+using GsaGH.Parameters;
+
+using OasysUnits.Units;
+
+using Xunit;
```
[the remark this answers is not in the document]
Comment (SandeepArup) [reply GsaGHTests/Helpers/GsaAPI/ModelFactoryTests.cs]: @psarras  all done. Kindly have a look

File: GsaGHTests/Helpers/GsaAPI/ModelFactoryTests.cs [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    [Fact]
+    public void ShouldCreateDefaultCaseWhenCaseCountIsZero() {
+      bool createDefaultCase = true;
+      ModelFactory.BuildAnalysisTask(_model.ApiModel, GsaAnalysisTasksFromSeedModel(), createDefaultCase);
+      Assert.Equal(2, _model.ApiModel.AnalysisTasks().First().Value.Cases.Count);
+    }
+
+    [Fact]
+    public void AnalysisCasesWillBeEmptyInAbsenceOfLoadCase() {
+      _model = new GsaModel();
```
Comment (psarras) [inline GsaGHTests/Helpers/GsaAPI/ModelFactoryTests.cs:97]: do you need this?

File: GsaGHTests/Helpers/GsaAPI/ModelFactoryTests.cs [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+      bool createDefaultCase = true;
+      ModelFactory.BuildAnalysisTask(_model.ApiModel, GsaAnalysisTasksFromSeedModel(), createDefaultCase);
```
Comment (psarras) [inline GsaGHTests/Helpers/GsaAPI/ModelFactoryTests.cs:99]: inline boolean

File: GsaGHTests/Helpers/GsaAPI/ModelFactoryTests.cs [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+
+    [Fact]
+    public void ShouldAddMissingTask() {
+      ModelFactory.BuildAnalysisTask(_model.ApiModel, GsaAnalysisTasksFromSeedModel());
+      Assert.Single(_model.ApiModel.AnalysisTasks());
+    }
+
+    [Fact]
+    public void ShouldNotCreateDefaultCaseWhenCaseCountNotZero() {
+      bool createDefaultCase = true;
```
Comment (psarras) [inline GsaGHTests/Helpers/GsaAPI/ModelFactoryTests.cs:83]: inline boolean

File: GsaGHTests/Helpers/GsaAPI/ModelFactoryTests.cs [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+      ModelFactory.BuildAnalysisTask(_model.ApiModel, GsaAnalysisTasksFromSeedModel(true), createDefaultCase);
+      Assert.Single(_model.ApiModel.AnalysisTasks().First().Value.Cases);
+    }
+
+    [Fact]
+    public void ShouldCreateDefaultCaseWhenCaseCountIsZero() {
+      bool createDefaultCase = true;
```
Comment (psarras) [inline GsaGHTests/Helpers/GsaAPI/ModelFactoryTests.cs:90]: inline boolean

**Review by psarras — CHANGES_REQUESTED** [review]
minor changes

File: GsaGHTests/Helpers/GsaAPI/ModelFactoryTests.cs (hunk 3 above)
Comment (SandeepArup) [reply GsaGHTests/Helpers/GsaAPI/ModelFactoryTests.cs:97]: Yes. As I want to create empty GSA model. One created in the constructor has load cases.

File: GsaGHTests/Helpers/GsaAPI/ModelFactoryTests.cs (hunk 4 above)
Comment (SandeepArup) [reply GsaGHTests/Helpers/GsaAPI/ModelFactoryTests.cs:99]: I had created It for clarity but have made it inline now

File: GsaGHTests/Helpers/GsaAPI/ModelFactoryTests.cs (hunk 5 above)
Comment (SandeepArup) [reply GsaGHTests/Helpers/GsaAPI/ModelFactoryTests.cs:83]: done

File: GsaGHTests/Helpers/GsaAPI/ModelFactoryTests.cs (hunk 6 above)
Comment (SandeepArup) [reply GsaGHTests/Helpers/GsaAPI/ModelFactoryTests.cs:90]: done

File: GsaGHTests/Helpers/GsaAPI/ModelFactoryTests.cs (hunk 3 above)
Comment (SandeepArup) [reply GsaGHTests/Helpers/GsaAPI/ModelFactoryTests.cs]: @psarras updated as per suggestion

**Review by DominikaLos — APPROVED** [review]

**Pull request closed without merging**